functions/src/on-create-bet.ts

```diff
@@ -38,37 +44,96 @@ export const onCreateBet = functions.firestore
       .doc(contractId)
       .update({ lastBetTime, lastUpdatedTime: Date.now() })
 
-    await notifyFills(bet, contractId, eventId)
-    await updateUniqueBettorsAndGiveCreatorBonus(
-      contractId,
-      eventId,
-      bet.userId
-    )
+    const userContractSnap = await firestore
+      .collection(`contracts`)
+      .doc(contractId)
+      .get()
+    const contract = userContractSnap.data() as Contract
+
+    if (!contract) {
+      log(`Could not find contract ${contractId}`)
+      return
+    }
+    await updateUniqueBettorsAndGiveCreatorBonus(contract, eventId, bet.userId)
+
+    const bettor = await getUser(bet.userId)
+    if (!bettor) return
+
+    await notifyFills(bet, contract, eventId, bettor)
+    await updateBettingStreak(bettor, bet, contract, eventId)
+
+    await firestore.collection('users').doc(bettor.id).update({ lastBetTime })
+  })
+
+const updateBettingStreak = async (
+  user: User,
+  bet: Bet,
+  contract: Contract,
+  eventId: string
+) => {
+  const betStreakResetTime = getPreviousBettingStreakResetTime()
+  // If they've already bet after the reset time, they've already gotten their streak bonus
+  if (user.lastBetTime ?? 0 > betStreakResetTime) return
+
+  const newBettingStreak = (user.currentBettingStreak ?? 0) + 1
```

Comment:
You could also actually include the reset logic here if you wanted: If the current bet time is over 24 hrs past the last betStreakResetTime, the current bet streak is just 1. Your approach with the cloud functions also works though.

Reply:
yeah that isn't a perfect solution because they need their streak to reset every day. Otherwise, if they don't make a bet for a few days their last streak will persist.